src/data/services/TagService.ts

```diff
@@ -0,0 +1,32 @@
+import { CommonServiceIds, IProjectPageService } from "azure-devops-extension-api";
+import { GitAnnotatedTag, GitRef, GitRestClient } from "azure-devops-extension-api/Git";
+import * as API from "azure-devops-extension-api";
+import * as SDK from "azure-devops-extension-sdk"
+import { PagedList } from "azure-devops-extension-api/WebApi";
+
+export class TagsService {
+    private gitClient: GitRestClient;
+    private projectId: string;
+
+    public async initialize(): Promise<void> {
+
+        this.gitClient = API.getClient(GitRestClient);
+        const projectService = await SDK.getService<IProjectPageService>(CommonServiceIds.ProjectPageService);
+        const project = await projectService.getProject();
+        this.projectId = project ? project?.id : "";
+
+    }
+
+    public async getAnnotedTags(repositoryId: string): Promise<GitAnnotatedTag[]> {
+        const refs = await this.getRefs(repositoryId, "tags");
+
+        const tags = await Promise.all(refs.map(async ref => await this.gitClient.getAnnotatedTag(this.projectId, repositoryId, ref.objectId)));
```

Comment:
At this time the GitRestClient doesn't support adding a limit to the request, even though it is mentioned on their docs: 
https://learn.microsoft.com/en-us/rest/api/azure/devops/git/refs/list?view=azure-devops-rest-7.0&tabs=HTTP

The default limit is set to 100.

Reply:
The `$top` parameter you refer to is made for pagination mostly. `getRefs` retursn a `PagedList` type, indicating there is a limit on the first result count. But I'd still suggest to have a hard limit before map into ~100 task, since it will produce a storm of backend calls.

Reply:
There is another problems with getAnnotatedTags.
`ref.objectId` refers to commit id in this case, and when multiple tags linked to the same commit `getAnnotatedTag` returns 404.
There is hardly any relevant information to gain from calling `getAnnotatedTag` you only need a name for drop down and some form of sorting by creation date - all this properties are provided by getRefs.

I advice to remove call to `getAnnotatedTags` and change name and signature of this method to reflect working with refs =)

Reply:
![image](https://github.com/dpetrovych/azure-devops-release-notes/assets/1765162/26bb8157-478c-46a9-b7d1-d80618cb3462)
Example of output for `product-availability-service` project

Reply:
Oh, good catch. :) I've removed getAnnotatedTags method, now it takes tags from getRefs.